lib/symboldatabase.cpp

```diff
@@ -2146,18 +2146,18 @@ void SymbolDatabase::validateExecutableScopes() const
 }
 
 namespace {
-    const Function* getFunctionForArgumentvariable(const Variable * const var, const std::vector<const Scope *>& functionScopes)
+    const Function* getFunctionForArgumentvariable(const Variable * const var)
     {
-        const std::size_t functions = functionScopes.size();
-        for (std::size_t i = 0; i < functions; ++i) {
-            const Scope* const scope = functionScopes[i];
-            const Function* const function = scope->function;
-            if (function) {
-                for (std::size_t arg=0; arg < function->argCount(); ++arg) {
-                    if (var==function->getArgumentVar(arg))
-                        return function;
+        if (const Scope* scope = var->nameToken()->scope()) {
```

Comment:
I am not sure if this works for a method where declaration and definition are in different scopes. something like:

```
class C {
    void dostuff(int x);
};

void C::dostuff(int x) {}
```

I might be mistaken but as far as I see the nameToken is in global scope but the function does not appear in the functionList of global scope?

Reply:
> Are you sure? There are unnamed arguments and if we ever want to get the function for such an argument it would crash here..

`getFunctionForArgumentvariable()` is called for every existing variable, therefore unnamed argument -> no variable.

Reply:
> I am not sure if this works for a method where declaration and definition are in different scopes. something like:
> 
> ```
> class C {
>     void dostuff(int x);
> };
> 
> void C::dostuff(int x) {}
> ```
> 
> I might be mistaken but as far as I see the nameToken is in global scope but the function does not appear in the functionList of global scope?

In this case, there is only one variable, and it has the scope `C::dostuff`. This is also true in case of a mismatch (e.g. declared as `x`, defined as `y`)

Reply:
ok